daft/datatype.py

```diff
@@ -443,10 +379,8 @@ def from_arrow_type(cls, arrow_type: pa.lib.DataType) -> DataType:
                     cls.from_arrow_type(arrow_type.storage_type),
                     metadata,
                 )
-        else:
-            # Fall back to a Python object type.
-            # TODO(Clark): Add native support for remaining Arrow types.
-            return cls.python()
```

Comment:
Note that removing this means that we **might** fail for certain arrow types, since we no longer fall back onto Python types.